### PR TITLE
FIX: improvements for user custom sections

### PR DESCRIPTION
**Pull request by lis2**
Improvements for this PR: https://github.com/discourse/discourse/pull/20057

What was fixed:
- [x] Use ember transitions instead of full reload
- [x] Link was inaccurately kept active
- [x] "+ save" renamed to just "save"
- [x] Render emojis in link name
- [x] UI to set icon
- [x] Delete link is trash icon instead of "x"
- [x] Add another link to on the left and rewording
- [x] Rename "link name" -> "name", "points to" ->  link
- [x] Move add section button to the bottom 

Screenshots:

Screenshot 2023-02-07 at 11 45 45 am
Screenshot 2023-02-07 at 11 46 03 am

Screenshot 2023-02-07 at 12 18 18 pm

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/sidebar/section-link.hbs [hunk 1]
```diff
@@ -7,7 +7,7 @@
       <a
         href={{@href}}
         rel="noopener noreferrer"
-        target={{or @target "_blank"}}
+        target="_blank"
```
Comment (lis2) [inline app/assets/javascripts/discourse/app/components/sidebar/section-link.hbs:10]: That was added in previous PR but once we moved to Ember route it is not required anymore https://github.com/discourse/discourse/pull/20057/files#diff-3f867149b0de3b82c2c5ddfee99564eee7b238eaf44775f4859da4677939d1a8R10

File: app/assets/javascripts/discourse/app/lib/sidebar/route-info-helper.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+export default class RouteInfoHelper {
```
Comment (lis2) [inline app/assets/javascripts/discourse/app/lib/sidebar/route-info-helper.js:1]: Fully extracted from app/assets/javascripts/discourse/app/lib/sidebar/custom-community-section-links.js

File: app/models/sidebar_url.rb [hunk 3]
```diff
@@ -1,8 +1,10 @@
 # frozen_string_literal: true
 
 class SidebarUrl < ActiveRecord::Base
-  validates :name, presence: true
-  validates :value, presence: true
+  validates :icon, presence: true, length: { maximum: 40 }
```
Comment (martin-brennan) [inline app/models/sidebar_url.rb:4]: Is the icon always required? If not no need for `presence` here

File: db/migrate/20230207042719_add_limits_to_sidebar_sections_and_sidebar_urls.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AddLimitsToSidebarSectionsAndSidebarUrls < ActiveRecord::Migration[7.0]
+  def change
+    change_column :sidebar_sections, :title, :string, limit: 30, null: false
+    change_column :sidebar_urls, :icon, :string, limit: 40, null: false
```
Comment (martin-brennan) [inline db/migrate/20230207042719_add_limits_to_sidebar_sections_and_sidebar_urls.rb:6]: Ah answered my own question about icon being required :+1:

**Review by martin-brennan — APPROVED** [review]
LGTM, nice!

**Pull request merged**